src/com/com_nrf24l01_core.cpp

```diff
@@ -0,0 +1,312 @@
+}
+
+auto NRF24L01Core::EnableAutoAck(DataPipe pipe_no) noexcept -> types::DriverStatus {
+  auto get_en_aa_reg = spi_protocol_->ReadRegister(reg::en_aa::REG_ADDR);
+  if (get_en_aa_reg.first != types::DriverStatus::OK) {
+    return get_en_aa_reg.first;
+  }
+  utilities::Byte en_aa_reg(get_en_aa_reg.second);
+  en_aa_reg.SetBit(static_cast<uint8_t>(pipe_no));
+  return (spi_protocol_->WriteRegister(reg::en_aa::REG_ADDR, en_aa_reg.Get()));
+}
+
+auto NRF24L01Core::ConfigAutoRetransmission(AutoRetransmissionDelay delay, AutoRetransmitCount count) noexcept -> types::DriverStatus {
+  auto get_setup_retr_reg = spi_protocol_->ReadRegister(reg::setup_retr::REG_ADDR);
+  if (get_setup_retr_reg.first != types::DriverStatus::OK) {
+    return get_setup_retr_reg.first;
+  }
+  utilities::Byte setup_retr_reg(get_setup_retr_reg.second);
+  setup_retr_reg.SetLowNibble(static_cast<uint8_t>(count));
+  setup_retr_reg.SetHighNibble(static_cast<uint8_t>(delay));
+  return (spi_protocol_->WriteRegister(reg::setup_retr::REG_ADDR, setup_retr_reg.Get()));
+}
+
+auto NRF24L01Core::MaskInterruptOnIntPin(MaskeableInterrupts interrupt) -> types::DriverStatus {
+  auto get_config_register = spi_protocol_->ReadRegister(reg::config::REG_ADDR);
+
+  ON_ERROR_RETURN(get_config_register.first);
+
+  utilities::Byte config_register(get_config_register.second);
+  config_register.SetBit(static_cast<std::uint8_t>(interrupt));
+
+  return (spi_protocol_->WriteRegister(reg::config::REG_ADDR, config_register.Get()));
+}
+
+auto NRF24L01Core::SetPipeAddress(DataPipe pipe_no, data_pipe_address pipe_addr) noexcept -> types::DriverStatus {
+  uint8_t register_addr = reg::rx_addr_p0::REG_ADDR + static_cast<std::uint8_t>(pipe_no);
+  std::vector<std::uint8_t> multibyte_addr;
+  std::uint8_t singlebyte_addr = 0;
+  auto return_value = types::DriverStatus::INPUT_ERROR;
+
+  switch (pipe_no) {
+    case DataPipe::TX_PIPE:
+    case DataPipe::RX_PIPE_0:
+    case DataPipe::RX_PIPE_1:
+      multibyte_addr.resize(addr_config::ADDR_WIDTH);
+      std::copy(pipe_addr.begin(),
+                pipe_addr.end(),
+                multibyte_addr.begin());
+      return_value = spi_protocol_->WriteRegister(register_addr, multibyte_addr);
+      break;
+    case DataPipe::RX_PIPE_2:
+    case DataPipe::RX_PIPE_3:
+    case DataPipe::RX_PIPE_4:
+    case DataPipe::RX_PIPE_5:
+      singlebyte_addr = pipe_addr[0];
+      return_value = spi_protocol_->WriteRegister(register_addr, singlebyte_addr);
+      break;
+  }
+  return return_value;
+}
+
+auto NRF24L01Core::GetPipeAddress(DataPipe pipe_no) noexcept -> std::pair<types::DriverStatus, data_pipe_address> {
+  uint8_t register_addr = reg::rx_addr_p0::REG_ADDR + static_cast<std::uint8_t>(pipe_no);
+  data_pipe_address addr = {0};
+  std::pair<types::DriverStatus, std::vector<std::uint8_t>> multibyte_addr;
+  std::pair<types::DriverStatus, std::uint8_t> singlebyte_addr;
+  types::DriverStatus return_value;
+
+  switch (pipe_no) {
```

Comment:
This is the same as before. Does this serve a purpose in keeping it a switch statement as opposed to a if else?

Reply:
see comment above.